### ISSUE TITLE
Cyclum circular plots look different from how they are supposed to be

### ISSUE THREAD

**Issue by No2Ross** [KChen-lab/cyclum#8]
Hi.

We've been running cyclum on some data and got some weird distrubtion across the round pseudotime plot

[Log2022_cyclum_fig_all 1.pdf](https://github.com/KChen-lab/Cyclum/files/14011470/Log2022_cyclum_fig_all.1.pdf)

We decided to test our version of cyclum on the H9 test data you provide in the notebook and find we get very different results for this dataset as well.

[round_plot_pseudotime_test.pdf](https://github.com/KChen-lab/Cyclum/files/14011471/round_plot_pseudotime_test.pdf)

Here's the code we've used to run the test data

Code:

# Add ../../ to path
import os,sys,inspect
current_dir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
parent_dir = os.path.dirname(current_dir)
sys.path.insert(0, parent_dir) 

import pandas as pd
from sklearn import preprocessing
import matplotlib.pyplot as plt

import numpy as np

import cyclum.tuning

cell_line = "H9"

raw_Y = pd.read_pickle('/mnt/data/project0015/CatM/cyclum/Cyclum/data/McDavid/h9_df.pkl').T

cpt = pd.read_pickle('/mnt/data/project0015/CatM/cyclum/Cyclum/data/McDavid/h9_cpt.pkl').values
print("Original dimesion %d cells x %d genes." % raw_Y.shape)
print(f"G0/G1 {sum(cpt == 'g0/g1')}, S {sum(cpt == 's')}, G2/M {sum(cpt == 'g2/m')}")

#expression_level = np.sum(2**raw_Y, axis=0)
#ind = np.argpartition(expression_level.values, -150)[-150:]
#genes = raw_Y.columns[np.argpartition(expression_level.values, -6)[-6:]]
#Y = raw_Y.loc[:, raw_Y.columns[ind]]
#Y.values[:, :] = preprocessing.scale(Y)

Y = preprocessing.scale(raw_Y)

N, D = Y.shape
print('After filtering %d Cells (instances) x %d Genes (features)'%(N, D))

model = cyclum.tuning.CyclumAutoTune(Y, max_linear_dims=3, 
                                     epochs=2000, rate=5e-4, verbose=100,
                                     encoder_width=[30, 20])

model.show_elbow()

import cyclum.models

model = cyclum.models.AutoEncoder(input_width=Y.shape[1],
                                  encoder_width=[30, 20], 
                                  encoder_depth=2,
                                 n_circular_unit=2,
                                 n_logistic_unit=0,
                                 n_linear_unit=0,
                                 n_linear_bypass=3,
                                 dropout_rate=0.1)

model.train(Y, epochs=1000, verbose=100, rate=2e-4)

pseudotime = model.predict_pseudotime(Y)

import cyclum.illustration
color_map = {"g0/g1": "red", "s": "green", "g2/m": "blue"}
fig =cyclum.illustration.plot_round_distr_color(pseudotime[:, 0], cpt.squeeze(), color_map)
fig.savefig('/mnt/data/project0015/CatM/cyclum/test/round_plot_pseudotime_test.pdf')

Here's the packages list as well:
_libgcc_mutex             0.1                        main  

_openmp_mutex             5.1                       1_gnu  

absl-py                   2.0.0                    pypi_0    pypi

anyio                     3.7.1                    pypi_0    pypi

argon2-cffi               23.1.0                   pypi_0    pypi

argon2-cffi-bindings      21.2.0                   pypi_0    pypi

astor                     0.8.1                    pypi_0    pypi

attrs                     23.1.0                   pypi_0    pypi

backcall                  0.2.0                    pypi_0    pypi

beautifulsoup4            4.12.2                   pypi_0    pypi

bleach                    6.0.0                    pypi_0    pypi

ca-certificates           2023.08.22           h06a4308_0  

certifi                   2022.12.7        py37h06a4308_0  

cffi                      1.15.1                   pypi_0    pypi

comm                      0.1.4                    pypi_0    pypi

cycler                    0.11.0                   pypi_0    pypi

cyclum-shaoheng           0.2a1                    pypi_0    pypi

debugpy                   1.7.0                    pypi_0    pypi

decorator                 5.1.1                    pypi_0    pypi

defusedxml                0.7.1                    pypi_0    pypi

entrypoints               0.4                      pypi_0    pypi

exceptiongroup            1.2.0                    pypi_0    pypi

fastjsonschema            2.19.0                   pypi_0    pypi

fonttools                 4.38.0                   pypi_0    pypi

gast                      0.2.2                    pypi_0    pypi

google-pasta              0.2.0                    pypi_0    pypi

grpcio                    1.59.3                   pypi_0    pypi

h5py                      3.8.0                    pypi_0    pypi

idna                      3.6                      pypi_0    pypi

importlib-metadata        6.7.0                    pypi_0    pypi

importlib-resources       5.12.0                   pypi_0    pypi

ipykernel                 6.16.2                   pypi_0    pypi

ipython                   7.34.0                   pypi_0    pypi

ipython-genutils          0.2.0                    pypi_0    pypi

ipywidgets                8.1.1                    pypi_0    pypi

jedi                      0.19.1                   pypi_0    pypi

jinja2                    3.1.2                    pypi_0    pypi

joblib                    1.3.2                    pypi_0    pypi

jsonschema                4.17.3                   pypi_0    pypi

jupyter                   1.0.0                    pypi_0    pypi

jupyter-client            7.4.9                    pypi_0    pypi

jupyter-console           6.6.3                    pypi_0    pypi

jupyter-core              4.12.0                   pypi_0    pypi

jupyter-server            1.24.0                   pypi_0    pypi

jupyterlab-pygments       0.2.2                    pypi_0    pypi

jupyterlab-widgets        3.0.9                    pypi_0    pypi

keras                     2.11.0                   pypi_0    pypi

keras-applications        1.0.8                    pypi_0    pypi

keras-preprocessing       1.1.2                    pypi_0    pypi

kiwisolver                1.4.5                    pypi_0    pypi

libedit                   3.1.20221030         h5eee18b_0  

libffi                    3.2.1             hf484d3e_1007  

libgcc-ng                 11.2.0               h1234567_1  

libgomp                   11.2.0               h1234567_1  

libstdcxx-ng              11.2.0               h1234567_1  

markdown                  3.4.4                    pypi_0    pypi

markupsafe                2.1.3                    pypi_0    pypi

matplotlib                3.5.3                    pypi_0    pypi

matplotlib-inline         0.1.6                    pypi_0    pypi

mistune                   3.0.2                    pypi_0    pypi

nbclassic                 1.0.0                    pypi_0    pypi

nbclient                  0.7.4                    pypi_0    pypi

nbconvert                 7.6.0                    pypi_0    pypi

nbformat                  5.8.0                    pypi_0    pypi

ncurses                   6.4                  h6a678d5_0  

nest-asyncio              1.5.8                    pypi_0    pypi

notebook                  6.5.6                    pypi_0    pypi

notebook-shim             0.2.3                    pypi_0    pypi

numpy                     1.21.6                   pypi_0    pypi

openssl                   1.1.1w               h7f8727e_0  

opt-einsum                3.3.0                    pypi_0    pypi

packaging                 23.2                     pypi_0    pypi

pandas                    1.3.5                    pypi_0    pypi

pandocfilters             1.5.0                    pypi_0    pypi

parso                     0.8.3                    pypi_0    pypi

pexpect                   4.9.0                    pypi_0    pypi

pickleshare               0.7.5                    pypi_0    pypi

pillow                    9.5.0                    pypi_0    pypi

pip                       22.3.1           py37h06a4308_0  

pkgutil-resolve-name      1.3.10                   pypi_0    pypi

prometheus-client         0.17.1                   pypi_0    pypi

prompt-toolkit            3.0.41                   pypi_0    pypi

protobuf                  3.20.3                   pypi_0    pypi

psutil                    5.9.6                    pypi_0    pypi

ptyprocess                0.7.0                    pypi_0    pypi

pycparser                 2.21                     pypi_0    pypi

pygments                  2.17.2                   pypi_0    pypi

pyparsing                 3.1.1                    pypi_0    pypi

pyrsistent                0.19.3                   pypi_0    pypi

python                    3.7.4                h265db76_1  

python-dateutil           2.8.2                    pypi_0    pypi

pytz                      2023.3.post1             pypi_0    pypi

pyzmq                     24.0.1                   pypi_0    pypi

qtconsole                 5.4.4                    pypi_0    pypi

qtpy                      2.4.1                    pypi_0    pypi

readline                  7.0                  h7b6447c_5  

scikit-learn              1.0.2                    pypi_0    pypi

scipy                     1.7.3                    pypi_0    pypi

send2trash                1.8.2                    pypi_0    pypi

setuptools                65.6.3           py37h06a4308_0  

six                       1.16.0                   pypi_0    pypi

sniffio                   1.3.0                    pypi_0    pypi

soupsieve                 2.4.1                    pypi_0    pypi

sqlite                    3.33.0               h62c20be_0  

tensorboard               1.15.0                   pypi_0    pypi

tensorflow                1.15.2                   pypi_0    pypi

tensorflow-estimator      1.15.1                   pypi_0    pypi

termcolor                 2.3.0                    pypi_0    pypi

terminado                 0.17.1                   pypi_0    pypi

threadpoolctl             3.1.0                    pypi_0    pypi

tinycss2                  1.2.1                    pypi_0    pypi

tk                        8.6.12               h1ccaba5_0  

tornado                   6.2                      pypi_0    pypi

traitlets                 5.9.0                    pypi_0    pypi

typing-extensions         4.7.1                    pypi_0    pypi

wcwidth                   0.2.12                   pypi_0    pypi

webencodings              0.5.1                    pypi_0    pypi

websocket-client          1.6.1                    pypi_0    pypi

werkzeug                  2.2.3                    pypi_0    pypi

wheel                     0.38.4           py37h06a4308_0  

widgetsnbextension        4.0.9                    pypi_0    pypi

wrapt                     1.16.0                   pypi_0    pypi

xz                        5.4.2                h5eee18b_0  

zipp                      3.15.0                   pypi_0    pypi

zlib                      1.2.13               h5eee18b_0
